src/uwtools/tests/drivers/test_jedi.py

```diff
@@ -0,0 +1,203 @@
+        )
+        assert (
+            call(src=Path("/path/to/foo"), dst=Path("/path/to/run/foo")) in filecopy.call_args_list
+        )
+
+
+def test_JEDI_files_linked(driverobj):
+    with patch.object(jedi, "symlink") as symlink:
+        driverobj._driver_config["run_dir"] = "/path/to/run"
+        driverobj.files_linked()
+        assert symlink.call_count == 2
+        assert (
+            call(target=Path("/path/to/baz"), linkname=Path("/path/to/run/bar/baz"))
+            in symlink.call_args_list
+        )
+        assert (
+            call(target=Path("/path/to/foo"), linkname=Path("/path/to/run/foo"))
+            in symlink.call_args_list
+        )
+
+
+def test_JEDI_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        files_copied=D,
+        files_linked=D,
+        runscript=D,
+        configuration_file=D,
+    ) as mocks:
+        driverobj.provisioned_run_directory()
+    for m in mocks:
+        mocks[m].assert_called_once_with()
+
+
+def test_JEDI_run_batch(driverobj):
+    with patch.object(driverobj, "_run_via_batch_submission") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_JEDI_run_local(driverobj):
+    driverobj._batch = False
+    with patch.object(driverobj, "_run_via_local_execution") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_JEDI_runscript(driverobj):
+    with patch.object(driverobj, "_runscript") as runscript:
+        driverobj.runscript()
+        runscript.assert_called_once()
+        args = ("envcmds", "envvars", "execution", "scheduler")
+        types = [list, dict, list, Slurm]
+        assert [type(runscript.call_args.kwargs[x]) for x in args] == types
+
+
+def test_JEDI_validate_only(caplog, driverobj):
+
+    @external
+    def file(path: Path):
+        yield "Mocked file task for %s" % path
+        yield asset(path, lambda: True)
+
+    logging.getLogger().setLevel(logging.INFO)
+    with patch.object(jedi, "file"):
+        with patch.object(jedi, "run") as run:
+            result = Mock(output="", success=True)
+            run.return_value = result
+            driverobj._driver_config["configuration_file"]["base_file"] = "/path/to/jedi.yaml"
+            driverobj.validate_only()
+            cmds = [
+                "module load some-module",
+                "module load jedi-module",
+                "time /path/to/qg_forecast.x --validate-only /path/to/jedi.yaml 2>&1",
+            ]
+            run.assert_called_once_with("20240201 18Z jedi validate_only", " && ".join(cmds))
+    assert regex_logged(caplog, "Config is valid")
+
+
+def test_JEDI_configuration_file(driverobj):
+    basecfg = {"foo": "bar"}
+    basefile = Path(driverobj._driver_config["configuration_file"]["base_file"])
+    with open(basefile, "w", encoding="utf-8") as f:
+        yaml.dump(basecfg, f)
+    cfgfile = Path(driverobj._driver_config["run_dir"]) / "input.yaml"
```

Comment:
Interesting, I changed this from "input.yaml" to "config.yaml" and replaced the same thing in the `jedi.py` ` configuration_file` task. After doing this it breaks with an error message of:

```
driverobj = <uwtools.drivers.jedi.JEDI object at 0x7faf0c333aa0>

    def test_JEDI_configuration_file(driverobj):
        basecfg = {"foo": "bar"}
        basefile = Path(driverobj._driver_config["configuration_file"]["base_file"])
        with open(basefile, "w", encoding="utf-8") as f:
            yaml.dump(basecfg, f)
        cfgfile = Path(driverobj._driver_config["run_dir"]) / "config.yaml"
>       assert not cfgfile.is_file()
E       AssertionError: assert not True
E        +  where True = <bound method Path.is_file of PosixPath('/tmp/pytest-of-Naureen.Bharwani/pytest-19/test_JEDI_configuration_file0/config.yaml')>()
E        +    where <bound method Path.is_file of PosixPath('/tmp/pytest-of-Naureen.Bharwani/pytest-19/test_JEDI_configuration_file0/config.yaml')> = PosixPath('/tmp/pytest-of-Naureen.Bharwani/pytest-19/test_JEDI_configuration_file0/config.yaml').is_file

src/uwtools/tests/drivers/test_jedi.py:191: AssertionError
```

Once I change the name back it's fine and I haven't found another place where `input.yaml` is used.

Reply:
Do `basefile` and `cfgfile` end up being the same file here? If so, is the `yaml.dump()` creating that file, such that the `assert` is failing because the file _does_ exist when it is expected not to yet?